### PR TITLE
Move Logging config to starlite.config

### PR DESCRIPTION
**Pull request by john-ingles**
Decouple logging config from logging module and move to starlite.config.logging

# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

It makes more sense to me to move the LoggingConfig to the starlite.config folder.  I made a new folder there for logging and readjusted imports across the repo to maintain the public api (so no need to modify docs).  The difference now is that the user doesn't have to look in the logging module to see the default logging config model and could even add their own/modify the existing one if they wish.

### CODE REVIEW

File: starlite/config/__init__.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
+from starlite.logging import QueueListenerHandler
```
Comment (cofin) [inline starlite/config/__init__.py:1]: Should this also be exported here?  it looks like it's still in the original `starlite.logging.__init__`

File: starlite/config/__init__.py [hunk 2]
```diff
@@ -14,4 +17,6 @@
     "StaticFilesConfig",
     "TemplateConfig",
     "CompressionConfig",
+    "LoggingConfig",
+    "QueueListenerHandler",
```
Comment (cofin) [inline starlite/config/__init__.py:21]: Same as above

**Review by cofin — CHANGES_REQUESTED** [review]
Looks good other than the question about the `QueueListenerHandler`.  I'll merge once we've resolved.

File: starlite/config/__init__.py (hunk 1 above)
Comment (john-ingles) [reply starlite/config/__init__.py:1]: I didn't move it to starlite.config because it's a function of the logging itself and unrelated to configuration

Comment (john-ingles) [reply starlite/config/__init__.py:1]: Wait, I think I see what you mean.  It doesn't need to be exported out of config.\_\_init\_\_.  I'll make a change

**Comment by cofin** [comment]
Looks good to me.  Nice work!

**Review by cofin — APPROVED** [review]

**Comment by cofin** [comment]
@all-contributors add @john-ingles code

**Pull request merged**